streams/test-utils/src/test/java/org/apache/kafka/streams/test/TestRecordTest.java

```diff
@@ -166,6 +167,9 @@ public void testProducerRecord() {
         final TestRecord<String, Integer> testRecord = new TestRecord<>(producerRecord);
         final TestRecord<String, Integer> expectedRecord = new TestRecord<>(key, value, headers, recordTime);
         assertEquals(expectedRecord, testRecord);
-        assertNotEquals(expectedRecord, producerRecord);
+        assertFalse(expectedRecord.getHeaders() == producerRecord.headers() &&
```

Comment:
Could we address this issue in separate PR? It seem to me this issue does not belong to "trivial cleanups" :)

Reply:
@chia7712 I see. Let me open a separate PR. :smile: